### PR TITLE
Added Cart and Wishlist functionality

### PR DESCRIPTION
**Pull request by deekshithmd**
Added cart and wishlist functionalities.
Check 
1.Cart
2.Wishlist
3.ProductCard
4.DataContext
5.DataReducer

### CODE REVIEW

File: src/services/services.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
 import axios from "axios";
 
-export const getProducts = async () => await axios.get("/api/products");
+export const getProducts = async () => axios.get("/api/products");
 
 export const getWishlist = async ({ encodedToken }) =>
-  await axios.get(`/api/user/wishlist`, {
+  axios.get(`/api/user/wishlist`, {
     headers: {
       authorization: encodedToken,
```
Comment (mayuresh-tech) [inline src/services/services.js:8]: Good usage of async!

File: src/components/productCard/ProductCard.jsx [hunk 2]
```diff
@@ -1,18 +1,77 @@
...
+  deleteWishlist,
+} from "../../services";
+
 const ProductCard = ({ product }) => {
+  const { data, dispatch, token } = useData();
+
+  const wish = data.wishlist.some((item) => item._id === product._id)
+    ? "fas fa-heart wishlisted"
+    : "far fa-heart";
+
```
Comment (mayuresh-tech) [inline src/components/productCard/ProductCard.jsx:18]: Suggestion:
You can add svgs so that UI is not dependent on websites for icons

**Review by mayuresh-tech — APPROVED** [review]
Good to merge!

File: src/pages/wishlist-page/Wishlist.jsx [hunk 3]
```diff
@@ -6,8 +6,10 @@ import {
   editCartlist,
 } from "../../services";
 import { useData } from "../../contexts";
+
 export default function Wishlist() {
   const { data, dispatch, token } = useData();
+
   async function addCart(product, tokens) {
     const responsew = await getCartlist({ encodedToken: tokens });
```
Comment (m26unkwn) [inline src/pages/wishlist-page/Wishlist.jsx]: Try to use Good Naming for variable

File: src/services/services.js [hunk 4]
```diff
@@ -1,16 +1,16 @@
 import axios from "axios";
 
-export const getProducts = async () => await axios.get("/api/products");
+export const getProducts = async () => axios.get("/api/products");
```
Comment (m26unkwn) [inline src/services/services.js:3]: Can you explain how is this working without using await inside the func.

**Review by m26unkwn — APPROVED** [review]
Code is clean and readable.
You can look at the following suggestion.

File: src/components/productCard/ProductCard.jsx (hunk 2 above)
Comment (deekshithmd) [reply src/components/productCard/ProductCard.jsx:18]: ok, Thank you for the suggestion

File: src/pages/wishlist-page/Wishlist.jsx (hunk 3 above)
Comment (deekshithmd) [reply src/pages/wishlist-page/Wishlist.jsx]: Noted

File: src/services/services.js (hunk 4 above)
Comment (deekshithmd) [reply src/services/services.js:3]: Wherever I am calling these api calls used async await

**Comment by deekshithmd** [comment]
> Code is clean and readable. You can look at the following suggestion.

Thank you for the review

**Pull request merged**